### PR TITLE
Refactor result cache for fewer tests that make requests

### PR DESCRIPTION
**Pull request by lukebarnard1**
This allows us to run the tests in `reporting-test.js` without making network requests. 

`handlers-test.js` still make requests but they are integration tests so this is expected.

### CODE REVIEW

File: src/reporting.js [hunk 1]
```diff
@@ -202,7 +197,13 @@ async function _generateReport(console, domain, mediaId, eventContentFile, opts)
     return result;
 }
 
-async function generateResult(console, eventContentFile, filePath, tempDir, script) {
+async function generateResult(console, httpUrl, eventContentFile, filePath, tempDir, script) {
```
Comment (dbkr) [inline src/reporting.js]: I'm finding it quite hard to keep track of what the different functions do here (especially the difference between `generateResult` and `generateReport `). `generateResult` == return cached scan or scan file on disk & cache result, `getReport ` == get from cache only, `generateReport ` == download file & run scan it? Might suggest renaming them to not rely on the difference between 'report' and 'result', and adding some doc to this method. Doc on the params would be appreciated too.

File: src/reporting.js [hunk 2]
```diff
@@ -81,8 +81,11 @@ const getReport = withTempDir(async function(console, domain, mediaId, eventCont
     const httpUrl = generateHttpUrl(baseUrl, domain, mediaId);
     const resultSecret = generateResultHash(httpUrl, eventContentFile);
 
-    const { clean, info } = resultCache[resultSecret]
-        || await generateReport(console, domain, mediaId, eventContentFile, opts);
+    if (!resultCache[resultSecret]) {
+        console.info(`File not scanned yet: domain = ${domain}, mediaId = ${mediaId}`);
+        return { scanned: false };
+    }
+    const { clean, info } = resultCache[resultSecret];
```
Comment (dbkr) [inline src/reporting.js]: If this isn't calling `generateReport` anymore, presumably it doesn't need to be `withTempDir`ed?

File: src/reporting.js [hunk 3]
```diff
@@ -202,7 +197,13 @@ async function _generateReport(console, domain, mediaId, eventContentFile, opts)
     return result;
 }
 
-async function generateResult(console, eventContentFile, filePath, tempDir, script) {
+async function generateResult(console, httpUrl, eventContentFile, filePath, tempDir, script) {
+    const resultSecret = generateResultHash(httpUrl, eventContentFile);
+    if (resultCache[resultSecret] !== undefined) {
+        console.info(`Result previously cached`);
+        return resultCache[resultSecret];
+    }
```
Comment (dbkr) [inline src/reporting.js]: Shouldn't this be using `getReport` rather than looking in the cache itself?

File: src/reporting.js [hunk 4]
```diff
@@ -81,8 +81,11 @@ const getReport = withTempDir(async function(console, domain, mediaId, eventCont
     const httpUrl = generateHttpUrl(baseUrl, domain, mediaId);
     const resultSecret = generateResultHash(httpUrl, eventContentFile);
 
-    const { clean, info } = resultCache[resultSecret]
-        || await generateReport(console, domain, mediaId, eventContentFile, opts);
+    if (!resultCache[resultSecret]) {
+        console.info(`File not scanned yet: domain = ${domain}, mediaId = ${mediaId}`);
+        return { scanned: false };
```
Comment (dbkr) [inline src/reporting.js]: Any reason to not just return `null`?

File: src/handlers.js [hunk 5]
```diff
@@ -79,7 +79,13 @@ async function encryptedScanReportHandler(req, res, next) {
 async function scanReportHandler(req, res, next, file) {
     const config = getConfig();
     const { domain, mediaId } = req.params;
-    const { clean, info } = await getReport(req.console, domain, mediaId, file, config.scan);
+    let result = await getReport(req.console, domain, mediaId, file, config.scan);
+
+    if (!result.scanned) {
+       result = await generateReport(req.console, domain, mediaId, file, config.scan);
+    }
+
```
Comment (dbkr) [inline src/handlers.js]: It seems odd that this is having to worry about checking the cache itself and calling another function if the cache missed.

File: src/reporting.js (hunk 2 above)
Comment (lukebarnard1) [reply src/reporting.js]: Yep, true.

**Comment by lukebarnard1** [comment]
Yay :smile: (mine too)

**Pull request merged**